tool/tctl/common/plugin/awsic.go

```diff
@@ -0,0 +1,208 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"maps"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+
+	pluginspb "github.com/gravitational/teleport/api/gen/proto/go/teleport/plugins/v1"
+	"github.com/gravitational/teleport/api/types"
+	apicommon "github.com/gravitational/teleport/api/types/common"
+	"github.com/gravitational/teleport/lib/utils"
+	toolcommon "github.com/gravitational/teleport/tool/common"
+)
+
+type awsICArgs struct {
+	cmd                  *kingpin.CmdClause
+	defaultOwners        []string
+	scimToken            string
+	scimURL              string
+	region               string
+	arn                  string
+	useSystemCredentials bool
+	userOrigin           string
+	userLabels           toolcommon.Labels
+	groupNameFilters     []string
+	accountNameFilters   []string
+	accountIDFilters     []string
+}
+
+func (a *awsICArgs) validate() error {
+	if !a.useSystemCredentials {
+		return trace.BadParameter("only AWS Local system credentials are supported")
+	}
+	return nil
+}
+
+// parseAWSICNameFilters validates that all elements of the supplied [names] slice
+// are valid regexes or globs and wraps them in [types.AWSICResourceFilter]s for
+// inclusion in a [types.PluginAWSICSettings].
+//
+// We are using a manual validator here rather than the canonical one defined
+// in the AWS IC integration itself, because those filter tools are not
+// available to OSS builds of tctl.
```

Comment:
Will it be a non-trivial migration to move that to oss?  I think we should reuse validation logics at minimum rather than having two different definition that risk going out of sync.

Reply:
I definitely want to move it, but I think it should be done in a separate PR.

Reply:
Added a `TODO` comment